### PR TITLE
[Merged by Bors] - feat: port Data.Int.SuccPred

### PR DESCRIPTION
**Pull request by Shreyas4991**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by semorrison** [comment]
I've added the `awaiting-review` label. Unless you do this, it won't appear on the [#queue4](https://bit.ly/3cKk3ld).

File: Mathlib/Data/Int/SuccPred.lean [hunk 1]
```diff
@@ -54,24 +54,24 @@ theorem succ_iterate (a : ℤ) : ∀ n, (succ^[n]) a = a + n
...
-    exact congr_arg _ (pred_iterate n)
+    exact congr_arg _ (pred_iterate a n)
 #align int.pred_iterate Int.pred_iterate
 
 instance : IsSuccArchimedean ℤ :=
-  ⟨fun a b h =>
+  ⟨fun {a b} h =>
     ⟨(b - a).toNat, by
-      rw [succ_eq_succ, succ_iterate, to_nat_sub_of_le h, ← add_sub_assoc, add_sub_cancel']⟩⟩
+      rw [succ_eq_succ, succ_iterate, toNat_sub_of_le h, ← add_sub_assoc, add_sub_cancel']⟩⟩
```
Comment (jcommelin) [inline Mathlib/Data/Int/SuccPred.lean:70]: Could you please add an `#align` statement for `to_nat_sub_of_le`? Probably somewhere in `Init.Data.Int`.

Comment (Shreyas4991) [reply Mathlib/Data/Int/SuccPred.lean:70]: I checked this and there already seems to be an align in `Mathlib/Data/Int/Order/Basic.lean`

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**